### DISCUSSION TITLE
When changing follow target node, if I try to change spring arm length it does nothing.

### DISCUSSION THREAD

**Discussion by deezgamezdev** [ramokz/phantom-camera#215] [Question / Help Wanted] [answered]
Hi,

Love the plugin by the way. I was just wondering about the orbitational rotation on targets. I wanted to make the camera angled just like in league of legends for example and be able to rotate around my player.

This was tricky but I was able to do it if I setup the look at target in the inspector and add an offset on the follow target. Then when I'm happy with the position I can delete de look at mode and reset my follow offset. This keeps the position of the camera where I want to and also without any bugs on rotating all around it.

Together with this setting I also linked the spring arm length to the mouse wheel so I can zoom out and in on the player.

Now, I'm making the same system but for vehicles, I setup a new camera just the same way as I setup for the player. But in this case we have alot of vehicles so I'm updating my follow target accordingly.

My problem is, when I update the follow target node, my camera position resets to 0 basically (only keeps the initial rotation) and my spring arm length "zoom function" doesn't do anything. I tested out by printing the spring arm value and it is being updated but it just doesn't make a difference after updating the follow target. If I start the game with the vehicle camera first, everything works correctly.

Can I get any help with this please ?
Thanks

https://github.com/ramokz/phantom-camera/assets/158978853/3260a1f2-226c-440a-94a6-f1f859459958

**Comment by deezgamezdev** [comment]
P.S: I dont leave the look at mode On because when I'm orbiting around the target, if the look at is On it stutters a bit due to the immediate action of the look at function.

**Comment by ramokz** [comment, marked as answer]
Assigning a `follow_target` to a Third Person `PCam` a few seconds into the game starts seems to work fine in the scene I'm testing in.

Two things to try out at first:

1. If you assign the vehicle as a target from the editor, does that produce the desired result?
2. If you disable all the `Collision Mask` layers, in your case just no. 1, does that change anything?

Also, just to sanity check, which addon version are you using?

**Comment by deezgamezdev** [comment]
Holy it worked !
If I assign the target in the editor and change it there is no problem.
And yes as soon as I disabled the collision masks for the PCam it worked like a charm. now it stays in the position I want and I can zoom without any issues.

Thank you so much.

But by the way, why this effect with the collisions ?

**Reply by ramokz** [reply]
No problem! Glad it works :slightly_smiling_face: 

It has to be due to there being a `PhysicsBody3D` of some kind in-between the actual `follow target` and the camera's initial position. So it basically thinks it's colliding against it the moment it starts to follow — hence the instant zoom.
Hard to say exactly what that node is exactly without digging into the scene tree for the vehicle and knowing what you're setting as the `follow target`.

The reason it works in the editor is that the `PCam3D` doesn't instantiate the `SpringArm3D`, which is what pushes itself inwards if a `PhysicsBody3D` is in-between it and the `PCam3D`, until you run the game. What the `PCam3D` does in the editor is merely a calculated value of its position, which doesn't take physics objects that might exist into account until you press play.

**Reply by deezgamezdev** [reply]
Hm, yes for the vehicles I'm using a RigidBody3D setup with some raycasts for custom vehicle physics but I will play around with the masks and see what I can achieve.

I think later I will want or have to make it collide with buildings and such but we will see along the way.

Thanks for the help and once again, amazing work with this plugin👏

**Reply by ramokz** [reply]
> I think later I will want or have to make it collide with buildings and such but we will see along the way.

That would be a simple case of just setting the various environment pieces (buildings) to e.g. collision mask layer 1 & 2, the player/vehicle to layer 1, and the `PCam3D` to just layer 2. Should mean you can get buildings to collide with the vehicle, the `PCam3D` to ignore the player/vehicle, but do collide with the environment. But obviously depends on the complexity of the project how well that layer setup would scale.
